src/test/java/emissary/util/IntegerUtilTest.java

```diff
@@ -0,0 +1,81 @@
+package emissary.util;
+
+import static org.junit.Assert.assertEquals;
+
+import java.math.BigInteger;
+
+import emissary.test.core.UnitTest;
+import org.junit.Test;
+
+public class IntegerUtilTest extends UnitTest {
+
+    @Test(expected = NumberFormatException.class)
+    public void testParseIntNullString() {
+        IntegerUtil.parseInt(null, 10, 0, 0);
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testParseIntEmptyString() {
+        IntegerUtil.parseInt("", 10, 0, 0);
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testParseIntRadixTooLow() {
+        IntegerUtil.parseInt("0", Character.MIN_RADIX - 1, 0, 1);
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testParseIntRadixTooHigh() {
+        IntegerUtil.parseInt("0", Character.MAX_RADIX + 1, 0, 1);
+    }
+
+    @Test
+    public void testParseIntLeadingPlus() {
+        assertEquals("Failed leading plus!", 1, IntegerUtil.parseInt("+1", 10, 0, 2));
+    }
+
+    @Test
+    public void testParseIntLeadingMinus() {
+        assertEquals("Failed leading plus!", -1, IntegerUtil.parseInt("-1", 10, 0, 2));
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testParseIntInvalidFirstCharacter() {
+        IntegerUtil.parseInt("*1", 10, 0, 2);
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testParseIntOnlyPlus() {
+        IntegerUtil.parseInt("+", 10, 0, 1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testParseIntNegativeStart() {
+        IntegerUtil.parseInt("0", 10, -1, 1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testParseIntNegativeLength() {
+        IntegerUtil.parseInt("0", 10, 0, -1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testParseIntInvalidStartLength() {
+        IntegerUtil.parseInt("0", 10, 1, 1);
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testExceedLimitSmall() {
+        final String s = BigInteger.valueOf(2).pow(31).add(BigInteger.valueOf(1)).toString(10);
+
+        IntegerUtil.parseInt(s, 10, 0, s.length());
+    }
+
+
+    @Test(expected = NumberFormatException.class)
+    public void testExceedLimitBig() {
+        final String s = BigInteger.valueOf(2).pow(31).add(BigInteger.valueOf(11)).toString(10);
```

Comment:
Not sure I understand how this differs from the previous test testExceedLimitSmall